### DISCUSSION TITLE
Discord Oauth is not working

### DISCUSSION THREAD

**Discussion by 0xWryth** [nextauthjs/next-auth#3340] [Help]
### Description 🐜

I am trying to use DiscordProvider to connect with Discord OAuth.
I am successfully getting Discord OAuth authorization page, and am successfully redirected to `localhost:3000`, but my `session` is currently `unauthorized` and I am not signed in.

Toggling `debug` to true in `[...nextauth].js`  show me a correct `[next-auth][debug][OAUTH_CALLBACK_RESPONSE]` informations, but session doesn't change and return `{data: null, status: "unauthenticated"}`.

I tried signing in with Github and did not encounter any problem.

### Is this a bug in your own project?

No

### How to reproduce ☕️

- Clone [next-auth-example](https://codesandbox.io/s/next-auth-example-1kktb).
- Renaming `.env.local.example` to `.env`
- Add `DISCORD_ID` and `DISCORD_SECRET` in `.env` file.
- Add the following lines to `[...nextauth].js`
```js
import DiscordProvider from "next-auth/providers/discord";

export default NextAuth({
  providers: [
    ...
    DiscordProvider({
      clientId: process.env.DISCORD_ID,
      clientSecret: process.env.DISCORD_SECRET,
    }),
  ],
  ...
}
```

### Environment 🖥

System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    Memory: 1.91 GB / 15.92 GB
  Binaries:
    Node: 16.8.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 7.21.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (96.0.1054.34), ChromiumDev (97.0.1072.13)
    Internet Explorer: 11.0.22000.120
  npmPackages:
    next: ^12.0.1 => 12.0.4
    next-auth: ^4.0.1 => 4.0.1
    react: ^17.0.2 => 17.0.2

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
did you configured the redirect url correctly in discord?

**Comment by 0xWryth** [comment]
I did set up the redirect URL as the following:

![image](https://user-images.githubusercontent.com/8791716/144483816-c1efd3b6-4449-4a45-bd9c-97e47d42e2a5.png)

**Reply by Subi** [reply]
Did you ever figure out the problem? Trying to set up Discord Authentication now and running into a similar issue

**Reply by 0xWryth** [reply]
Sorry for the late answer, clearing my browser cache seemed to fixed the issue.
I was using Edge Dev when I encountered this issue, Edge, Chrome, or Firefox seemed to work perfectly.

**Comment by balazsorban44** [comment]
This is likely an issue with your setup. I tried it multiple times, and multiple accounts, and it works correctly.
